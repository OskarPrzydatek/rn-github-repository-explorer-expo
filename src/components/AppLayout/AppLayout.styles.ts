import { StyleSheet } from 'react-native';

export const appLayoutStyles = StyleSheet.create({
  appLayout: {
    flex: 1,
    paddingHorizontal: 16,
    marginTop: 64,
    gap: 16,
  },
});
