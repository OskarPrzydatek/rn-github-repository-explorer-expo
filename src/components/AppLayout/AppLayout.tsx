import React, { ReactNode } from 'react';

import { SafeAreaView } from 'react-native-safe-area-context';

import { appLayoutStyles } from '@/components/AppLayout/AppLayout.styles';

interface IAppLayout {
  children: ReactNode;
}

export const AppLayout = ({ children }: IAppLayout) => {
  return (
    <SafeAreaView style={appLayoutStyles.appLayout}>{children}</SafeAreaView>
  );
};
