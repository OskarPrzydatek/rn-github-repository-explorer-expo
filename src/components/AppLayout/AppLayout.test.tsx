import { render } from '@testing-library/react-native';
import { View } from 'react-native';

import { AppLayout } from '@/components/AppLayout/AppLayout';

describe('AppLayout', () => {
  it('should match snapshot', () => {
    const tree = render(
      <AppLayout>
        <View />
      </AppLayout>,
    );

    expect(tree).toMatchSnapshot();
  });
});
