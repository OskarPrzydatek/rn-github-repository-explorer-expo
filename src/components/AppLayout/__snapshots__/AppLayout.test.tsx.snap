// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
      "gap": 16,
      "marginTop": 64,
      "paddingHorizontal": 16,
    }
  }
>
  <View />
</RNCSafeAreaView>
`;
